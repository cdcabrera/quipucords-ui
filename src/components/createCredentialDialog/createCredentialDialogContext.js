import { reduxActions, reduxTypes, storeHooks } from '../../redux';

/**
 * Return credential dialog response
 *
 * @param {object} options
 * @param {Function} options.useSelector
 * @returns {*}
 */
const useCredential = ({ useSelector: useAliasSelector = storeHooks.reactRedux.useSelector } = {}) =>
  useAliasSelector(({ credentials }) => credentials?.dialog, {});

/**
 * Return an updated source. Display relative toast messaging after wizard closes.
 *
 * @param {object} options
 * @param {Function} options.addCredential
 * @param {Function} options.updateCredential
 * @param {Function} options.useDispatch
 * @returns {(function(*, *): void)|*}
 */
const useOnSubmitCredential = ({
  addCredential = reduxActions.credentials.addCredential,
  updateCredential = reduxActions.credentials.updateCredential,
  useDispatch: useAliasDispatch = storeHooks.reactRedux.useDispatch
} = {}) => {
  const dispatch = useAliasDispatch();

  return (id, data) => {
    if (id) {
      updateCredential(id, data)(dispatch);
    } else {
      addCredential(data)(dispatch);
    }
  };
};

/**
 * Functions for handling state updates for credentials.
 *
 * @param {object} options
 * @param {Function} options.useDispatch
 * @returns {{onEdit: Function, onHide: Function, onAdd: Function}}
 */
const useOnUpdateCredential = ({ useDispatch: useAliasDispatch = storeHooks.reactRedux.useDispatch } = {}) => {
  const dispatch = useAliasDispatch();

  /**
   * On edit show the credential in dialog
   *
   * @event onEdit
   * @param {object|*} credential
   */
  const onEdit = credential => {
    dispatch({
      type: reduxTypes.credentials.EDIT_CREDENTIAL_SHOW,
      credential
    });
  };

  /**
   * Hide the credential dialog
   *
   * @event onHide
   */
  const onHide = () => {
    dispatch({
      type: reduxTypes.credentials.UPDATE_CREDENTIAL_HIDE
    });
  };

  /**
   * on add show the credential dialog
   *
   * @event onAdd
   * @param {string|*} credentialType
   */
  const onAdd = credentialType => {
    dispatch([
      {
        type: reduxTypes.credentials.CREATE_CREDENTIAL_SHOW,
        credentialType
      }
    ]);
  };

  return {
    onEdit,
    onHide,
    onAdd
  };
};

const context = {
  useCredential,
  useOnSubmitCredential,
  useOnUpdateCredential
};

export { context as default, context, useCredential, useOnSubmitCredential, useOnUpdateCredential };
