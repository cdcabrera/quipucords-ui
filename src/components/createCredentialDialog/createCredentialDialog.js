import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  ButtonVariant,
  EmptyState,
  EmptyStateIcon,
  Form,
  Spinner,
  Title,
  ValidatedOptions
} from '@patternfly/react-core';
import { useCredential, useOnSubmitCredential, useOnUpdateCredential } from './createCredentialDialogContext';
import { Modal } from '../modal/modal';
import { DropdownSelect, SelectDirection } from '../dropdownSelect/dropdownSelect';
import { FormGroup } from '../form/formGroup';
import { TextInput } from '../form/textInput';
import { FormState } from '../formState/formState';
import { formHelpers } from '../form/formHelpers';
import { apiTypes } from '../../constants/apiConstants';
import { translate } from '../i18n/i18n';

/**
 * Available method options
 *
 * @type {string[]}
 */
const becomeMethodTypeOptions = ['sudo', 'su', 'pbrun', 'pfexec', 'doas', 'dzdo', 'ksu', 'runas'];

/**
 * Generate authentication type options.
 *
 * @type {{title: Function | React.ReactNode, value: string, selected: boolean}[]}
 */
const authenticationTypeOptions = [
  {
    title: () => translate('form-dialog.label', { context: ['option', 'sshKey'] }),
    value: 'sshKey'
  },
  {
    title: () => translate('form-dialog.label', { context: ['option', 'token'] }),
    value: 'token'
  },
  {
    title: () => translate('form-dialog.label', { context: ['option', 'usernamePassword'] }),
    value: 'usernamePassword',
    selected: true
  }
];

// ToDo: "sshpassphrase" could be "ssh_passphrase" per the api-spec, investigate since the prior GUI code used "sshpassphrase"
/**
 * Create or edit a credential.
 *
 * @fires onSetAuthType
 * @fires onCancel
 * @fires onSubmit
 * @fires onValidateForm
 * @param {object} props
 * @param {Array} props.authenticationOptions
 * @param {Array} props.becomeMethodOptions
 * @param {Function} props.t
 * @param {Function} props.useCredential
 * @param {Function} props.useOnSubmitCredential
 * @param {Function} props.useOnUpdateCredential
 * @returns {React.ReactNode|null}
 */
const CreateCredentialDialog = ({
  authenticationOptions,
  becomeMethodOptions,
  t,
  useCredential: useAliasCredential,
  useOnSubmitCredential: useAliasOnSubmitCredential,
  useOnUpdateCredential: useAliasOnUpdateCredential
}) => {
  const [authType, setAuthType] = useState();
  const { show, edit, credential = {}, credentialType, pending } = useAliasCredential();
  const { onHide } = useAliasOnUpdateCredential();
  const submitCredential = useAliasOnSubmitCredential();

  useEffect(() => {
    switch (credentialType) {
      case 'openshift':
        setAuthType('token');
        break;
      case 'network':
      case 'satellite':
      case 'vcenter':
      default:
        setAuthType('usernamePassword');
        break;
    }
  }, [credentialType]);

  if (!credentialType) {
    return null;
  }

  /**
   * Reset form fields on auth type selection.
   *
   * @event onSetAuthType
   * @param {object} event
   * @param {*} event.value
   * @param {Function} handleOnEventCustom
   */
  const onSetAuthType = ({ value }, handleOnEventCustom) => {
    handleOnEventCustom([
      {
        name: apiTypes.API_QUERY_TYPES.AUTH_TOKEN,
        value: undefined
      },
      {
        name: apiTypes.API_QUERY_TYPES.PASSWORD,
        value: undefined
      },
      {
        name: apiTypes.API_QUERY_TYPES.SSH_KEYFILE,
        value: undefined
      },
      {
        name: apiTypes.API_QUERY_TYPES.SSH_PASSPHRASE,
        value: undefined
      }
    ]);

    setAuthType(value);
  };

  /**
   * Hide the dialog
   *
   * @event onCancel
   * @returns {*}
   */
  const onCancel = () => onHide();

  /**
   * Submit form state to add or update a credential.
   *
   * @event onSubmit
   * @param {object} formState
   * @param {object} formState.values
   */
  const onSubmit = ({ values = {} } = {}) => {
    const updatedValues = {};
    Object.entries(values)
      .filter(([, value]) => value !== undefined)
      .forEach(([key, value]) => {
        updatedValues[key] = value;
      });

    submitCredential(updatedValues);
  };

  /**
   * Form validator, return an error object against field names using form state.
   *
   * @event onValidateForm
   * @param {object} formState
   * @param {object} formState.values
   * @returns {{ssh_keyfile: boolean, password: boolean, name: boolean, auth_token: boolean, username: boolean}}
   */
  const onValidateForm = ({ values = {} } = {}) => ({
    [apiTypes.API_QUERY_TYPES.NAME]: formHelpers.isEmpty(values[apiTypes.API_QUERY_TYPES.NAME]),
    [apiTypes.API_QUERY_TYPES.AUTH_TOKEN]:
      authType === 'token' && formHelpers.isEmpty(values[apiTypes.API_QUERY_TYPES.AUTH_TOKEN]),
    [apiTypes.API_QUERY_TYPES.SSH_KEYFILE]:
      (authType === 'sshKey' && formHelpers.isEmpty(values[apiTypes.API_QUERY_TYPES.SSH_KEYFILE])) ||
      (authType === 'sshKey' && !formHelpers.isFilePath(values[apiTypes.API_QUERY_TYPES.SSH_KEYFILE])),
    [apiTypes.API_QUERY_TYPES.PASSWORD]:
      authType === 'usernamePassword' && formHelpers.isEmpty(values[apiTypes.API_QUERY_TYPES.PASSWORD]),
    [apiTypes.API_QUERY_TYPES.USERNAME]:
      authType === 'usernamePassword' && formHelpers.isEmpty(values[apiTypes.API_QUERY_TYPES.USERNAME])
  });

  /**
   * Pass form state and render field(s) common to all credential types.
   *
   * @param {object} formState
   * @param {object} formState.errors
   * @param {object} formState.touched
   * @param {object} formState.values
   * @param {Function} formState.handleOnEvent
   * @param {Function} formState.handleOnEventCustom
   * @returns {React.ReactNode}
   */
  const renderCommonFields = ({ errors, touched, values, handleOnEvent, handleOnEventCustom }) => (
    <React.Fragment>
      <FormGroup
        label={t('form-dialog.label', { context: ['name', 'create-credential'] })}
        error={touched[apiTypes.API_QUERY_TYPES.NAME] && errors[apiTypes.API_QUERY_TYPES.NAME]}
        errorMessage="You must enter a credential name"
      >
        <TextInput
          id="cred_name"
          name={apiTypes.API_QUERY_TYPES.NAME}
          value={values.name}
          placeholder="Enter a name for the credential"
          onChange={handleOnEvent}
          onClear={handleOnEvent}
          maxLength={64}
          validated={
            touched[apiTypes.API_QUERY_TYPES.NAME] && errors[apiTypes.API_QUERY_TYPES.NAME]
              ? ValidatedOptions.error
              : ValidatedOptions.default
          }
        />
      </FormGroup>
      {values[apiTypes.API_QUERY_TYPES.CREDENTIAL_TYPE] === 'network' && (
        <FormGroup label="Authentication Type">
          <DropdownSelect
            isInline={false}
            onSelect={event => onSetAuthType(event, handleOnEventCustom)}
            options={authenticationOptions}
            selectedOptions={authType}
          />
        </FormGroup>
      )}
      <FormGroup
        label="Username"
        error={touched[apiTypes.API_QUERY_TYPES.USERNAME] && errors[apiTypes.API_QUERY_TYPES.USERNAME]}
        errorMessage="You must enter a username"
      >
        <TextInput
          name={apiTypes.API_QUERY_TYPES.USERNAME}
          value={values[apiTypes.API_QUERY_TYPES.USERNAME]}
          placeholder="Enter Username"
          onChange={handleOnEvent}
          onClear={handleOnEvent}
          validated={
            touched[apiTypes.API_QUERY_TYPES.USERNAME] && errors[apiTypes.API_QUERY_TYPES.USERNAME]
              ? ValidatedOptions.error
              : ValidatedOptions.default
          }
        />
      </FormGroup>
    </React.Fragment>
  );

  /**
   * Pass form state and render authentication field(s) against authentication type.
   *
   * @param {object} formState
   * @param {object} formState.errors
   * @param {object} formState.touched
   * @param {object} formState.values
   * @param {Function} formState.handleOnEvent
   * @returns {React.ReactNode}
   */
  const renderAuthFields = ({ errors, touched, values, handleOnEvent } = {}) => {
    switch (authType) {
      case 'sshKey':
        return (
          <React.Fragment>
            <FormGroup
              key="ssh_keyfile"
              label="SSH Key File"
              error={touched[apiTypes.API_QUERY_TYPES.SSH_KEYFILE] && errors[apiTypes.API_QUERY_TYPES.SSH_KEYFILE]}
              errorMessage="Please enter the full path to the SSH Key File"
            >
              <TextInput
                name={apiTypes.API_QUERY_TYPES.SSH_KEYFILE}
                value={values[apiTypes.API_QUERY_TYPES.SSH_KEYFILE]}
                placeholder="Enter a SSH key file path"
                onChange={handleOnEvent}
                onClear={handleOnEvent}
                validated={
                  touched[apiTypes.API_QUERY_TYPES.SSH_KEYFILE] && errors[apiTypes.API_QUERY_TYPES.SSH_KEYFILE]
                    ? ValidatedOptions.error
                    : ValidatedOptions.default
                }
              />
            </FormGroup>
            <FormGroup key="sshpassphrase" label="Passphrase">
              <TextInput
                name={apiTypes.API_QUERY_TYPES.SSH_PASSPHRASE}
                type="password"
                value={values[apiTypes.API_QUERY_TYPES.SSH_PASSPHRASE]}
                placeholder="optional"
                onChange={handleOnEvent}
                onClear={handleOnEvent}
              />
            </FormGroup>
          </React.Fragment>
        );
      case 'token':
        return (
          <FormGroup
            key="auth_token"
            label="Token"
            error={touched[apiTypes.API_QUERY_TYPES.AUTH_TOKEN] && errors[apiTypes.API_QUERY_TYPES.AUTH_TOKEN]}
            errorMessage="You must enter a token"
          >
            <TextInput
              name={apiTypes.API_QUERY_TYPES.AUTH_TOKEN}
              value={values[apiTypes.API_QUERY_TYPES.AUTH_TOKEN]}
              placeholder="Enter token"
              onChange={handleOnEvent}
              onClear={handleOnEvent}
              validated={
                touched[apiTypes.API_QUERY_TYPES.AUTH_TOKEN] && errors[apiTypes.API_QUERY_TYPES.AUTH_TOKEN]
                  ? ValidatedOptions.error
                  : ValidatedOptions.default
              }
            />
          </FormGroup>
        );
      case 'usernamePassword':
      default:
        return (
          <FormGroup
            key="password"
            label="Password"
            error={touched[apiTypes.API_QUERY_TYPES.PASSWORD] && errors[apiTypes.API_QUERY_TYPES.PASSWORD]}
            errorMessage="You must enter a password"
          >
            <TextInput
              name={apiTypes.API_QUERY_TYPES.PASSWORD}
              type="password"
              value={values[apiTypes.API_QUERY_TYPES.PASSWORD]}
              placeholder="Enter password"
              onChange={handleOnEvent}
              onClear={handleOnEvent}
              validated={
                touched[apiTypes.API_QUERY_TYPES.PASSWORD] && errors[apiTypes.API_QUERY_TYPES.PASSWORD]
                  ? ValidatedOptions.error
                  : ValidatedOptions.default
              }
            />
          </FormGroup>
        );
    }
  };

  /**
   * Pass form state and render network credential field(s).
   *
   * @param {object} formState
   * @param {object} formState.values
   * @param {Function} formState.handleOnEvent
   * @returns {React.ReactNode}
   */
  const renderNetworkFields = ({ values, handleOnEvent } = {}) => {
    if (values[apiTypes.API_QUERY_TYPES.CREDENTIAL_TYPE] !== 'network') {
      return null;
    }

    return (
      <React.Fragment>
        <FormGroup key="become_method" label="Become Method">
          <DropdownSelect
            name={apiTypes.API_QUERY_TYPES.BECOME_METHOD}
            isInline={false}
            onSelect={handleOnEvent}
            options={becomeMethodOptions}
            selectedOptions={values[apiTypes.API_QUERY_TYPES.BECOME_METHOD]}
            direction={SelectDirection.up}
          />
        </FormGroup>
        <FormGroup key="become_user" label="Become User">
          <TextInput
            name={apiTypes.API_QUERY_TYPES.BECOME_USER}
            type="text"
            value={values[apiTypes.API_QUERY_TYPES.BECOME_USER]}
            placeholder="optional"
            onChange={handleOnEvent}
            onClear={handleOnEvent}
          />
        </FormGroup>
        <FormGroup key="become_password" label="Become Password">
          <TextInput
            name={apiTypes.API_QUERY_TYPES.BECOME_PASSWORD}
            type="password"
            value={values[apiTypes.API_QUERY_TYPES.BECOME_PASSWORD]}
            placeholder="optional"
            onChange={handleOnEvent}
            onClear={handleOnEvent}
          />
        </FormGroup>
      </React.Fragment>
    );
  };

  return (
    <FormState
      setValues={{ ...credential, [apiTypes.API_QUERY_TYPES.CREDENTIAL_TYPE]: credentialType }}
      validateOnMount={false}
      validate={onValidateForm}
      onSubmit={onSubmit}
    >
      {({ handleOnSubmit, isValid, values, ...options } = {}) => (
        <Modal
          isOpen={show}
          showClose
          onClose={onCancel}
          header={<Title headingLevel="h4">{edit ? `View Credential` : 'Add Credential'}</Title>}
          actions={[
            <Button key="save" onClick={handleOnSubmit} isDisabled={!isValid}>
              {t('form-dialog.label', { context: ['submit', edit && 'edit', 'create-credential'] })}
            </Button>,
            <Button key="cancel" variant={ButtonVariant.secondary} autoFocus={edit} onClick={onCancel}>
              {t('form-dialog.label', { context: 'cancel' })}
            </Button>
          ]}
        >
          <Form isHorizontal onSubmit={handleOnSubmit}>
            {pending && (
              <EmptyState className="quipucords-empty-state">
                <EmptyStateIcon icon={Spinner} />
                <Title headingLevel="h3">
                  {t('form-dialog.empty-state', { context: ['title', 'create-scan', 'pending'] })}
                </Title>
              </EmptyState>
            )}
            {!pending && (
              <React.Fragment>
                <FormGroup label="Source Type">
                  <TextInput
                    type="text"
                    isReadOnly
                    value={t('form-dialog.label', { context: values[apiTypes.API_QUERY_TYPES.CREDENTIAL_TYPE] })}
                  />
                </FormGroup>
                {renderCommonFields({ values, ...options })}
                {renderAuthFields({ values, ...options })}
                {renderNetworkFields({ values, ...options })}
              </React.Fragment>
            )}
          </Form>
        </Modal>
      )}
    </FormState>
  );
};

CreateCredentialDialog.propTypes = {
  authenticationOptions: PropTypes.array,
  becomeMethodOptions: PropTypes.array,
  t: PropTypes.func,
  useCredential: PropTypes.func,
  useOnSubmitCredential: PropTypes.func,
  useOnUpdateCredential: PropTypes.func
};

CreateCredentialDialog.defaultProps = {
  authenticationOptions: authenticationTypeOptions,
  becomeMethodOptions: becomeMethodTypeOptions,
  t: translate,
  useCredential,
  useOnSubmitCredential,
  useOnUpdateCredential
};

export {
  CreateCredentialDialog as default,
  CreateCredentialDialog,
  authenticationTypeOptions,
  becomeMethodTypeOptions
};
