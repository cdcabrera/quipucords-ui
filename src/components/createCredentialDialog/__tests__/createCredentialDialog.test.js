import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { ConnectedCreateCredentialDialog, CreateCredentialDialog } from '../createCredentialDialog';

describe('CreateCredentialDialog Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a connected component', () => {
    const store = generateEmptyStore({ credentials: { update: { show: true } }, viewOptions: {} });

    const component = shallow(<ConnectedCreateCredentialDialog />, { context: { store } });
    expect(component.dive()).toMatchSnapshot('connected');
  });

  it('should render a non-connected component', () => {
    const props = {
      show: false
    };

    const component = mount(<CreateCredentialDialog {...props} />);
    expect(component).toMatchSnapshot('non-connected');
  });
});
