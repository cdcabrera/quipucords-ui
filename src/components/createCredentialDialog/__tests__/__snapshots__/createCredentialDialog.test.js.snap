// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCredentialDialog Component should render a connected component: connected 1`] = `
<div
  class="pf-c-backdrop"
>
  <div
    class="pf-l-bullseye"
  >
    <div
      aria-describedby="pf-modal-part-2"
      aria-label="t(modal.aria-label-default)"
      aria-labelledby="pf-modal-part-0"
      aria-modal="true"
      class="pf-c-modal-box pf-m-align-top"
      data-ouia-component-id="OUIA-Generated-Modal-1"
      data-ouia-component-type="PF4/ModalContent"
      data-ouia-safe="true"
      id="pf-modal-part-0"
      role="dialog"
      style="--pf-c-modal-box--m-align-top--spacer: 5%;"
    >
      <button
        aria-disabled="false"
        aria-label="Close"
        class="pf-c-button pf-m-plain"
        data-ouia-component-id="OUIA-Generated-Modal-1-ModalBoxCloseButton"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 352 512"
          width="1em"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          />
        </svg>
      </button>
      <header
        class="pf-c-modal-box__header"
      >
        <h4
          class="pf-c-title pf-m-md"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
        >
          Add Credential
        </h4>
      </header>
      <div
        class="pf-c-modal-box__body"
        id="pf-modal-part-2"
      >
        <div
          class="container-fluid"
        >
          <form
            class="form-horizontal"
          >
            <div
              class="form-group"
            >
              <label
                class="col-sm-5 control-label"
              >
                Source Type
              </label>
              <div
                class="col-sm-7"
              >
                <input
                  class="quipucords-form-control form-control"
                  readonly=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group"
            >
              <label
                class="col-sm-5 control-label"
              >
                Credential Name
              </label>
              <div
                class="col-sm-7"
              >
                <input
                  class="quipucords-form-control form-control"
                  placeholder="Enter a name for the credential"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group"
            >
              <label
                class="col-sm-5 control-label"
              >
                Authentication Type
              </label>
              <div
                class="col-sm-7"
              >
                <div
                  class="quipucords-dropdownselect  dropdown btn-group"
                  name="auth-type-select"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="form-control dropdown-toggle btn btn-default"
                    id="auth-type-select"
                    role="button"
                    type="button"
                  >
                    <span>
                      Username and Password
                    </span>
                     
                    <span
                      class="caret"
                    />
                  </button>
                  <ul
                    aria-labelledby="auth-type-select"
                    class="dropdown-menu"
                    role="menu"
                  >
                    <li
                      class=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        role="menuitem"
                        tabindex="-1"
                      >
                        SSH Key
                      </a>
                    </li>
                    <li
                      class="active"
                      role="presentation"
                    >
                      <a
                        href="#"
                        role="menuitem"
                        tabindex="-1"
                      >
                        Username and Password
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="form-group"
            >
              <label
                class="col-sm-5 control-label"
              >
                Username
              </label>
              <div
                class="col-sm-7"
              >
                <input
                  class="form-control"
                  placeholder="Enter Username"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group"
            >
              <label
                class="col-sm-5 control-label"
              >
                Password
              </label>
              <div
                class="col-sm-7"
              >
                <input
                  class="form-control"
                  placeholder="Enter Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </form>
        </div>
      </div>
      <footer
        class="pf-c-modal-box__footer"
      >
        <button
          aria-disabled="true"
          class="pf-c-button pf-m-primary pf-m-disabled"
          data-ouia-component-id="OUIA-Generated-Button-primary-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          disabled=""
          type="button"
        >
          t(form-dialog.label_submit, {"context":"create-credential"})
        </button>
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-secondary"
          data-ouia-component-id="OUIA-Generated-Button-secondary-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          t(form-dialog.label, {"context":"cancel"})
        </button>
      </footer>
    </div>
  </div>
</div>
`;

exports[`CreateCredentialDialog Component should render a non-connected component: non-connected 1`] = `null`;
